jwst/source_catalog/source_catalog_step.py

```diff
@@ -119,4 +119,9 @@ def process(self, input_model):
                 model.meta.source_catalog = os.path.basename(cat_filepath)
```

Comment:
Since we're saving the name of the source catalog in a header keyword of the image product, I feel that for completeness we should also save the name of the segmentation map file. We'll need to create a new keyword for this in `core.schema.yaml`, perhaps called `SEGMFILE` (similar to `SCATFILE`) and give it a datamodel name of `segmentation_map`. So would need to add a line here (or below) to populate that keyword.